crates/uv-trampoline/src/bounce.rs

```diff
@@ -459,139 +460,123 @@ fn close_handles(si: &STARTUPINFOA) {
-        let mut msg = MaybeUninit::<MSG>::uninit();
-        GetMessageA(msg.as_mut_ptr(), 0, 0, 0);
-        WaitForInputIdle(child_handle, INFINITE);
+        // End the launcher's "app starting" cursor state.
+        PostMessageA(None, 0, None, None).unwrap_or_else(|_| {
+            eprintln!("Failed to post a message to specified window.");
+        });
+        if GetMessageA(msg.as_mut_ptr(), None, 0, 0) != TRUE {
+            eprintln!("Failed to retrieve posted window message.");
+        }
+        // Proxy the child's input idle event.
+        if WaitForInputIdle(child_handle, INFINITE) != 0 {
+            eprintln!("Failed to wait for input from window.");
+        }
+        // Signal the process input idle event by creating a window and pumping
+        // sent messages. The window class isn't important, so just use the
+        // system "STATIC" class.
         let hwnd = CreateWindowExA(
-            0,
-            c"STATIC".as_ptr() as *const _,
-            c"uv Python Trampoline".as_ptr() as *const _,
-            0,
+            WINDOW_EX_STYLE(0),
+            s!("STATIC"),
+            s!("uv Python Trampoline"),
+            WINDOW_STYLE(0),
             0,
             0,
             0,
             0,
             HWND_MESSAGE,
-            0,
-            0,
-            null(),
+            None,
+            None,
+            None,
         );
-        PeekMessageA(msg.as_mut_ptr(), hwnd, 0, 0, 0);
-        DestroyWindow(hwnd);
+        // Process all sent messages and signal input idle.
+        _ = PeekMessageA(msg.as_mut_ptr(), hwnd, 0, 0, PEEK_MESSAGE_REMOVE_TYPE(0));
+        DestroyWindow(hwnd).unwrap_or_else(|_| {
+            print_last_error_and_exit("Failed to destroy temporary window.");
+        });
     }
 }
 
 pub fn bounce(is_gui: bool) -> ! {
-    unsafe {
-        let child_cmdline = make_child_cmdline();
+    let child_cmdline = make_child_cmdline();
 
-        let mut si = MaybeUninit::<STARTUPINFOA>::uninit();
-        GetStartupInfoA(si.as_mut_ptr());
-        let si = si.assume_init();
+    let mut si = MaybeUninit::<STARTUPINFOA>::uninit();
+    unsafe { GetStartupInfoA(si.as_mut_ptr()) }
+    let si = unsafe { si.assume_init() };
 
-        let child_handle = spawn_child(&si, child_cmdline);
-        let job = make_job_object();
-        expect_result(AssignProcessToJobObject(job, child_handle), 0, || {
-            String::from("Error from AssignProcessToJobObject")
-        });
-
-        // (best effort) Close all the handles that we can
-        close_handles(&si);
+    let child_handle = spawn_child(&si, child_cmdline);
+    let job = make_job_object();
 
-        // (best effort) Switch to some innocuous directory so we don't hold the original
-        // cwd open.
-        if let Some(tmp) = getenv(c"TEMP") {
-            SetCurrentDirectoryA(tmp.as_ptr() as *const _);
-        } else {
-            SetCurrentDirectoryA(c"c:\\".as_ptr() as *const _);
```

Comment:
Note, the `c:\\` approach was removed as it's not part of [launcher.c](https://github.com/pypa/distlib/blob/master/PC/launcher.c#L736), plus the sysroot drive could vary on a system.

getenv(c"TEMP") was changed to use std instead as it had bugs when parsing the value of TEMP on my system probably due to W vs A calls but didn't dig too much, as a result `getenv` was deleted.